velox/functions/lib/Re2Functions.cpp

```diff
@@ -116,6 +116,48 @@ bool re2Extract(
   }
 }
 
+void checkEscape(bool condition) {
+  if (!condition) {
+    VELOX_USER_FAIL(
+        "Escape character must be followed by '%%', '_' or the escape character itself");
+  }
+}
+
+StringView likePatternToRe2(StringView pattern, char escapeChar) {
+  std::string regex;
+  regex.append("^");
+  bool escaped = false;
+  for (const char* c = pattern.begin(); c < pattern.end(); ++c) {
+    checkEscape(!escaped || *c == '%' || *c == '_' || *c == escapeChar);
+    if (!escaped && (*c == escapeChar)) {
+      escaped = true;
+    } else {
+      switch (*c) {
+        case '%':
+          regex.append(escaped ? "%" : ".*");
+          escaped = false;
+          break;
+        case '_':
+          regex.append(escaped ? "_" : ".");
+          escaped = false;
+          break;
+        case '\\':
+        case '^':
+        case '$':
+        case '.':
+        case '*':
+          regex.append("\\");
+        default:
+          regex.append(1, c[0]);
+          escaped = false;
+      }
+    }
+  }
+  checkEscape(!escaped);
+  regex.append("$");
+  return StringView(regex);
```

Comment:
Hmm... This won't work if regex is more than 12 characters long and cannot be stored inlined in StringView. Return std::string instead.